datarobot_batch_scoring/batch_scoring.py

```diff
@@ -412,23 +416,53 @@ def __iter__(self):
                                'we lost {} records'.format(
                                    batch.id, len(batch.data)))
                 continue
-            hook = partial(self._response_callback, batch=batch)
 
             if self.fast_mode:
                 chunk_formatter = fast_to_csv_chunk
             else:
                 chunk_formatter = slow_to_csv_chunk
 
-            data = chunk_formatter(batch.data, batch.fieldnames)
-            self._ui.debug('batch {} transmitting {} bytes'
-                           .format(batch.id, len(data)))
-            yield requests.Request(
-                method='POST',
-                url=self.endpoint,
-                headers=self.headers,
-                data=data,
-                auth=(self.user, self.api_token),
-                hooks={'response': hook})
+            todo = [batch]
+            while todo:
+                batch = todo.pop(0)
+                data = chunk_formatter(batch.data, batch.fieldnames)
+                if len(data) < self.max_batch_size:
```

Comment:
Most of the necessary logic exists in this function. https://github.com/datarobot/batch-scoring/blob/master/datarobot_batch_scoring/utils.py#L420 We could probably a test for max-size here